### ISSUE TITLE
HAProxy tests have a dict-ordering reliance

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8286]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8286|
|Type|enhancement|
|Created|2016-04-10 12:47:51Z|
|Branch|https://github.com/twisted/twisted/tree/haproxy-dict-ordering-8286|

```
[FAIL]
Traceback (most recent call last):
  File "/buildslave/ubuntu12.04-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/protocols/haproxy/test/test_parser.py", line 57, in test_keywords
    self.check("aleph=one:bet=two")
  File "/buildslave/ubuntu12.04-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/protocols/haproxy/test/test_parser.py", line 36, in check
    self.assertEqual(unparseEndpoint(*parseEndpoint(input)), input)
  File "/buildslave/ubuntu12.04-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 437, in assertEqual
    super(_Assertions, self).assertEqual(first, second, msg)
  File "/usr/lib/python2.7/unittest/case.py", line 511, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/lib/python2.7/unittest/case.py", line 504, in _baseAssertEqual
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: 'bet=two:aleph=one' != 'aleph=one:bet=two'

twisted.protocols.haproxy.test.test_parser.UnparseEndpointTests.test_keywords
```

<details><summary>Searchable metadata</summary>

```
trac-id__8286 8286
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_haproxy_dict_ordering_8286 branches/haproxy-dict-ordering-8286
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1460292471869109 1460292471869109
changetime__1460346577626835 1460346577626835
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [47202]) Branching to haproxy-dict-ordering-8286.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Builders spun, looking good.

The Fedora 23 test failures are due to [#8279](https://github.com/twisted/twisted/issues/8279)#ticket and are unrelated.

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_3"></a>|@mithrandi set owner to @hawkowl|
|-|-|

Looks good, please merge :)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [47208]) Merge haproxy-dict-ordering-8286: Fix a test that relied on dict ordering

Author: hawkowl
Reviewer: mithrandi
Fixes: [#8286](https://github.com/twisted/twisted/issues/8286)
